Clarify highlightsFromPoint() example and functionality (#43804)
mdn/content: files/en-us/web/api/highlightregistry/highlightsfrompoint/index.md

#### HTML

The markup includes a {{htmlelement("p")}} element and a {{htmlelement("section")}} element. The `<section>` serves as the output area where the content of the double-clicked highlights are displayed.

```html live-sample___highlights-from-point-example
<h1>highlightsFromPoint() demo</h1>
<h2>Highlightable content</h2>
<p class="highlightable-text">
  When you select a section of text and then press "h" on the keyboard, the text
  you selected will be given a custom highlight. Multiple highlights will be
  colored yellow, red, and blue, in that order. When you double-click on a
  highlighted section of text, that section will be output at the bottom of the
  UI. If multiple highlights overlap the section, you'll see multiple text
  sections. Select a portion of text, and then press the "h" key. The selected
  text gets a custom highlight, colored yellow, red, or blue, in that order.
  After the third highlight, each new one replaces the oldest, cycling through
  the colors in the same order. Next, double-click any highlighted text. The
  highlighted text will appear in the output. If multiple highlights overlap a

>   section, you'll see multiple text sections in the output.

Reviewer: There's some duplicate text here before the "Select a portion of text..." sentence
```suggestion
Select a portion of text, and then press the "h" key. The selected
  text gets a custom highlight, colored yellow, red, or blue, in that order.
  After the third highlight, each new one replaces the oldest, cycling through
  the colors in the same order. Next, double-click any highlighted text. The
  highlighted text will appear in the output. If multiple highlights overlap a
  section, you'll see multiple text sections in the output.
```

> ### Output custom highlights applied at the mouse pointer position

Reviewer: might want to update this title as well:

starting verb: Displaying or retrieving?

"Retrieving highlights at the mouse pointer position"?

> This example demonstrates using the `highlightsFromPoint()` method to return the content of all custom highlights located at the mouse pointer coordinates of a user's double-click. In this example, multiple custom highlights can be created on a paragraph of text, including overlapping ones. When the user presses the <kbd>h</kbd> key after selecting some text, a new {{domxref("Highlight")}} is named and registered, supporting up to three custom highlights at a time. When the user double-clicks on the paragraph, the content of all the double-clicked highlights, if any, are written to the page.

Reviewer: - splitting into two paras
- also suggesting rephrasing in 2nd para to make it extra clear
  - "double-clicks a highlight" not the para
```suggestion
This example demonstrates how to use the `highlightsFromPoint()` method to return the content of all custom highlights located at the mouse pointer coordinates of a user's double-click.

In this example, multiple custom highlights can be created on a paragraph of text, and the highlights can overlap. When the user presses the <kbd>h</kbd> key after selecting some text, a new {{domxref("Highlight")}} is named and registered. This example supports up to three custom highlights at a time. When the user double-clicks within the highlighted area, the content of all highlights at that point, if any, is displayed in the output area.
```

#### HTML

> The markup includes a {{htmlelement("p")}} element and a {{htmlelement("section")}} element. The `<section>` is included as the container where the highlighted text fragments will be displayed.

Reviewer: ```suggestion
The markup includes a {{htmlelement("p")}} element and a {{htmlelement("section")}} element. The `<section>` serves as the output area where the content of the double-clicked highlights are displayed.
```

  multiple highlights overlap the section, you'll see multiple text sections

>   outputted.

Reviewer: - Making the language here imperative so that it is clear that this is the instruction to follow to play around with this UI
- I guess we should also mention what'll happen if readers make more than three selections+"h"
```suggestion
Select a portion of text, and then press the "h" key. The selected text gets a custom highlight, colored yellow, red, or blue, in that order. After the third highlight, each new one replaces the oldest, cycling through the colors in the same order.

Next, double-click any highlighted text. The text in that highlight will appear in the output. If multiple highlights overlap a section, you'll see multiple text sections in the output.
```

</p>

> <h2>Text of clicked highlights</h2>

Reviewer: or this?
```suggestion
<h2>Text in double-clicked highlights</h2>
```

<section></section>
```

#### CSS

> In the CSS, we define styling for three custom highlights named `highlight1`, `highlight2`, and `highlight3`. We select each custom highlight by passing its name into the {{cssxref("::highlight()")}} pseudo-element, making their backgrounds semi-transparent yellow, red, or blue, respectively, with areas of combined colors where the semi-transparent backgrounds of the highlights overlap.

Reviewer: ```suggestion
In the CSS, we define styling for three custom highlights named `highlight1`, `highlight2`, and `highlight3`. We target each custom highlight using the {{cssxref("::highlight()")}} pseudo-element, making their backgrounds semi-transparent yellow, red, and blue, respectively. Where highlights overlap, the semi-transparent backgrounds combine to show a mixed color.
```

  border: 2px solid #dddddd;
  border-radius: 5px;
}
```

#### JavaScript

> This example has two distinct areas of functionality. We first have to enable the creation of custom highlights when the user clicks <kbd>h</kbd> after having selected some text. We then have to enable writing highlighted content to the page when the user double clicks on one or more custom highlights.

Reviewer: ```suggestion
This example has two distinct areas of functionality. We first enable the creation of custom highlights when the user clicks the <kbd>h</kbd> key after selecting some text. We then enable writing the highlighted content to the page when the user double-clicks one or more custom highlights.
```

{{EmbedLiveSample("highlights-from-point-example", "100%", "600")}}

> To create a highlight, press <kbd>h</kbd> after selecting some text. You can create up to three highlights. Double click on the highlights you created, preferably where they overlap, to write the content of the clicked highlights to the page.

Reviewer: Do you want to remove this since the instruction is part of the demo itself?

Author: I prefer not to have readers rely on reading an example's text (or code comments). For several reasons. Mostly cognitive a11y. Most examples don't contain instructions, so the user may not think to look there for instructions. Also, I am not sure if this content is translated when pages are translated.

Reviewer: All good points! Thanks
